dvc/remote/base.py

```diff
@@ -697,6 +697,9 @@ def gc(self, named_cache):
             if checksum in used:
                 continue
             path_info = self.checksum_to_path_info(checksum)
+            if self.is_dir_checksum(checksum):
+                if callable(getattr(self, "_get_unpacked_dir_path_info")):
```

Comment:
Probably don't need `callable` :slightly_smiling_face: Also, this `getattr` will keep working normally even if we remove `_get_unpacked_dir_path_info` at some point. So your implementation is correct, but maybe we could clean it up a bit. How about we mimic the other `unpacked_dir` logic that we have in the code and define `_remove_unpacked_dir` method in base(as a noop) and in local, so that we could simply:
```
if self.is_dir_checksum(checksum):
    self._remove_unpacked_dir(checksum)
```
here, and won't have to do `getattr` tricks?

Also, please add a test for this to `tests/func/test_gc.py`. Something similar to the reproduction script in the issue would be just right :slightly_smiling_face: